src/useLocalStorage.ts

```diff
@@ -69,7 +69,7 @@ const useLocalStorage = <T>(
         // localStorage can throw. Also JSON.stringify can throw.
       }
     },
-    [key, setState]
+    [key, state, setState]
```

Comment:
Alternative fix can be `setState((cur) => {...}` to avoid accessing the state declared outside the callback.